src/main/java/app/model/agents/WallFollow/WallFollowAgent.java

```diff
@@ -546,6 +588,32 @@ public boolean noObstacleDetectedFromPos(double rayAngle, int distFromAgentVerte
         return true;
     }
 
+    public void markWallAsCovered()
+    {
+        // TODO implement pheromone smell here?
+        BooleanCell forwardCell = getAgentNeighbourBasedOnAngle(direction.getAngle());
+        if (forwardCell.getX() == cellGraph.getAgentPos().getX())
+        {
+            horizontalWallsCovered.add(forwardCell.getX());
+        }
+        if (forwardCell.getY() == cellGraph.getAgentPos().getY())
+        {
+            verticalWallsCovered.add(forwardCell.getY());
+        }
+    }
+
+    private boolean foundUnexploredWallToFollow()
+    {
+        BooleanCell forwardCell = getAgentNeighbourBasedOnAngle(direction.getAngle());
+        if (!noWallDetected(direction.getAngle()) && forwardCell.getObstacle()
+                && ((!horizontalWallsCovered.contains(forwardCell.getX()) && direction.getX() == 0)
+                || (!verticalWallsCovered.contains(forwardCell.getY()) && direction.getY() == 0)))
+        {
```

Comment:
consider simplifying to :
```java

return (!noWallDetected(direction.getAngle()) && forwardCell.getObstacle()
           && ((!horizontalWallsCovered.contains(forwardCell.getX()) && direction.getX() == 0)
           || (!verticalWallsCovered.contains(forwardCell.getY()) && direction.getY() == 0)))
```
As returning a boolean evaluation will always give you the same as an if true/ else false